src/main/java/picard/util/LiftoverUtils.java

```diff
@@ -0,0 +1,255 @@
+        }
+
+        //not currently supported
+        if (source.isIndel() && !source.isBiallelic()){
+            return null;
+        }
+
+        List<Allele> origAlleles = new ArrayList<>(source.getAlleles());
+        VariantContextBuilder vcb = new VariantContextBuilder(source);
+        vcb.chr(target.getContig());
+
+        final boolean addToStart = source.isIndel() && target.getStart() > 1;
+
+        final int start = target.getStart() - (addToStart ? 1 : 0);
+        vcb.start(start);
+
+        final int stop = target.getEnd() - (addToStart ? 1 : 0);
+        vcb.stop(stop);
+
+        vcb.alleles(reverseComplementAlleles(origAlleles, target, refSeq, source.isIndel(), addToStart));
+        if (source.isIndel()){
+            leftAlignVariant(vcb, start, stop, vcb.getAlleles(), refSeq);
+        }
+
+        vcb.genotypes(fixGenotypes(source.getGenotypes(), origAlleles, vcb.getAlleles()));
+
+        return vcb;
+    }
+
+    private static List<Allele> reverseComplementAlleles(List<Allele> originalAlleles, Interval target, ReferenceSequence refSeq, boolean isBiAllelicIndel, boolean addToStart){
+        final List<Allele> alleles = new ArrayList<>();
+
+        for (final Allele oldAllele : originalAlleles) {
+            alleles.add(LiftoverUtils.reverseComplement(oldAllele, target, refSeq, isBiAllelicIndel, addToStart));
+        }
+
+        return alleles;
+    }
+
+    private static Allele reverseComplement(Allele oldAllele, Interval target, ReferenceSequence referenceSequence, boolean isBiAllelicIndel, boolean addToStart){
+
+        if (oldAllele.isSymbolic()){
+            return oldAllele;
+        }
+        else if (isBiAllelicIndel){
+            // target.getStart is 1-based, reference bases are 0-based
+            final StringBuilder alleleBuilder = new StringBuilder(target.getEnd() - target.getStart() + 1);
+
+            if (addToStart) alleleBuilder.append((char) referenceSequence.getBases()[target.getStart() - 2]);
+            alleleBuilder.append(SequenceUtil.reverseComplement(oldAllele.getBaseString().substring(1, oldAllele.length())));
+            if (!addToStart) alleleBuilder.append((char) referenceSequence.getBases()[target.getEnd() - 1]);
+
+            return Allele.create(alleleBuilder.toString(), oldAllele.isReference());
+        }
+        else {
+            return Allele.create(SequenceUtil.reverseComplement(oldAllele.getBaseString()), oldAllele.isReference());
+        }
+    }
+
+    protected static GenotypesContext fixGenotypes(final GenotypesContext originals, List<Allele> originalAlleles, List<Allele> newAlleles) {
+        // optimization: if nothing needs to be fixed then don't bother
+        if (originalAlleles.equals(newAlleles)) {
+            return originals;
+        }
+
+        final GenotypesContext fixedGenotypes = GenotypesContext.create(originals.size());
+        for (final Genotype genotype : originals) {
```

Comment:
This is where the map (from old allele to new allele) would help. But you removed it. now you need to mess around with indexes and stuff. Consider the case of many genotypes...would it not be better to have a map?

The alleles in the genotype must be part of the VC's alleles, so the checking (for symbolic but not no-call).

Reply:
The reason the map was awkward (or at least the original usage of it was), if because translating genotypes wasnt a one-step process.  You need to do the initial reverse complement, and then maybe left align, etc.  We ended up with multiple maps, updating the genotypes multiple times (the thing i wanted to avoid).  It's entirely possible this approach could have been made to work, but it seemed messy when I way working on it.

The reference and alt alleles are in fact an ordered list.  Genotypes (at least in the raw file), just point to allele index, and dont actually 'know' what that allele means.  If I could someone prevent any parsing of genotypes at all until after we fixed up the alleles that would be even better.  Dealing with updating alleles with genotypes at every change is an unnecessary pain.  

I changed this to make a one-time map in this method based on originalAlleles and newAlleles, instead of using indexOf() each time we iterate all the genotypes.

Reply:
I get that, I'm saying this allele-swapping process would be made easier if that's not how it worked.  If VariantContext could do lazy parsing of the genotypes (i.e. so they remain 0/1) until after we swap around the ref/alt alleles then this translation wouldnt matter one bit.  In any case, that's not what it does.

In the updated PR I done a one-time loop to make this old/new allele map, and translate alleles for each genotype based on this.